ejb-txn-remote-call/README.adoc

```diff
@@ -0,0 +1,942 @@
+
+The http://jbossts.blogspot.com/2018/01/narayana-periodic-recovery-of-xa.html[recovery manager]
+normally executes the recovery processing in periodically every 2 minutes.
+When the recovery process is started the resources at the remote server (on the callee side)
+are committed.
+
+You may speed up the process and invoke the recovery process manually by accessing
+the port where recovery manager listener listens at. The recovery listener was enabled
+for this purpose by cli command, see `${PATH_TO_QUICKSTART_DIR}/ejb-txn-remote-call/client/extensions/remote-configuration.cli`.
+For starting the recovery processing send `SCAN` command to socket at `localhost:4712`.
+
+[source]
+----
+telnet localhost 4712
+Trying 127.0.0.1...
+Connected to localhost.
+Escape character is '^]'.
+SCAN
+DONE
+Connection closed by foreign host.
+----
+
+Steps for observe the processing
+
+. Invoke the endpoint
++
+[source,options="nowrap"]
+----
+curl -s http://localhost:8080/client/remote-outbound-fail-stateless
+----
++
+. Check the server logs
+.. The `server1` will contain error
++
+[source,options="nowrap"]
+----
+ARJUNA016036: commit on < formatId=131077, gtrid_length=35, bqual_length=36, tx_uid=..., node_name=server1, branch_uid=..., subordinatenodename=null, eis_name=unknown eis name > (Subordinate XAResource at remote+http://localhost:8180) failed with exception $XAException.XA_RETRY: javax.transaction.xa.XAException: WFTXN0029: The peer threw an XA exception
+----
++
+. The `server2` or `server3` will contain error
++
+[source,options="nowrap"]
+----
+ARJUNA016036: commit on < formatId=131077, gtrid_length=35, bqual_length=43, tx_uid=..., node_name=server1, branch_uid=..., subordinatenodename=server2, eis_name=unknown eis name > (org.jboss.as.quickstarts.ejb.mock.MockXAResource@731ae22) failed with exception $XAException.XAER_RMFAIL: javax.transaction.xa.XAException
+----
++
+. Verify number of commits done for the test XAResource at the server side
++
+[source,options="nowrap"]
+----
+curl -s http://localhost:8180/server/commits
+curl -s http://localhost:8280/server/commits
+----
++
+. Start recovery processing by entering `SCAN` string with command `telnet localhost 4712` at terminal
++
+[NOTE]
+====
+If the telnet is not available at your machine is recommended to be installed for you may proceed
+further with this guide.
+====
++
+.. After the recovery is processed the server log at `server2` or `server3` should contain warning
++
+[source,options="nowrap"]
+----
+WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016114: Could not load org.jboss.as.quickstarts.ejb.mock.MockXAResource will try to get XAResource from the recovery helpers
+----
++
+. Verify the number of commits done for the test XAResource at the server side. The commit count should be increased by one.
++
+[source,options="nowrap"]
+----
+curl -s http://localhost:8180/server/commits
+curl -s http://localhost:8280/server/commits
```

Comment:
I got the same results as when I ran the commands above on line 435:

```bash
[mmusgrov@dev1 wildfly.latest] (master) $ curl -s http://localhost:8180/server/commits
["host: dev1.ncl.jboss.com/192.168.0.33, jboss node name: server2","4"][mmusgrov@dev1 wildfly.latest] (master) $
[mmusgrov@dev1 wildfly.latest] (master) $ curl -s http://localhost:8280/server/commits
["host: dev1.ncl.jboss.com/192.168.0.33, jboss node name: server3","0"][mmusgrov@dev1 wildfly.latest] (master) $
```

Does that mean the test failed or does it mean that recovery already happened or something else?